### DISCUSSION TITLE
Keycloak not redirecting back to the Spring boot App -

### DISCUSSION THREAD

**Discussion by saravanad** [ch4mpy/spring-addons#110] [Q&A] [answered]
Hi @ch4mpy ,

I have keycloak configured properly in SSL certificate. 

Using the bff project, i am able to log in using keycloak. However after login, the redirect URL is not taking me back to bff app.

It still has the Keycloak base URL instead of bff app base URL.

For Example.  the redirect URL in the browser after login is 

https://XXX.YYY.com/login/oauth2/code/keycloak-confidential-user?state=GQfm5392nWe9cuHiZyTbzYKZMPndNR-qG-TwJWyw-FI%3D&session_state=ca49f6df-5924-41d9-88e4-4ef71e127995&code=b26b0d1f-73e0-46b7-92ab-49e234c5aae1.ca49f6df-5924-41d9-88e4-4ef71e127995.86626423-fa07-40fc-8e8a-af8c555b2687

whereas it should be 

http://localhost:8080/login/oauth2/code/keycloak-confidential-user?state=GQfm5392nWe9cuHiZyTbzYKZMPndNR-qG-TwJWyw-FI%3D&session_state=ca49f6df-5924-41d9-88e4-4ef71e127995&code=b26b0d1f-73e0-46b7-92ab-49e234c5aae1.ca49f6df-5924-41d9-88e4-4ef71e127995.86626423-fa07-40fc-8e8a-af8c555b2687

Any specific pointers or how to address this issue?  Thanks in advance for your support

Thanks,
Saravana

**Comment by saravanad** [comment]
The Same base URL incorrect behavior is also observed for logout action as well.

**Comment by ch4mpy** [comment]
Which BFF app? Is your code available from somewhere like Github?

**Reply by saravanad** [reply, marked as answer]
BFF App which was part of this github repository. No the code is not available in Github.

However the issue is solved. I had incorrect configuration in keycloak. 

Thanks for your response.
